mate-panel/button-widget.c

```diff
@@ -158,50 +159,67 @@ button_widget_unset_surfaces (ButtonWidget *button)
 static void
 button_widget_reload_surface (ButtonWidget *button)
 {
-    GdkDisplay *display;
-    gint scale;
+	GdkDisplay *display;
+	gint scale;
 	button_widget_unset_surfaces (button);
 
+	button->priv->needs_move = FALSE;
 	if (button->priv->size <= 1 || button->priv->icon_theme == NULL)
 		return;
 
 	if (button->priv->filename != NULL &&
 	    button->priv->filename [0] != '\0') {
 		char *error = NULL;
-
+		/* icons findable in the icon theme can be handled by gtk directly*/
 		display = gdk_display_get_default ();
 		scale = gtk_widget_get_scale_factor (GTK_WIDGET (button));
-		if  (GDK_IS_X11_DISPLAY (display))
+		GtkIconTheme *icon_theme = gtk_icon_theme_get_default();
+		button->priv->surface =
+			gtk_icon_theme_load_surface (icon_theme,
+				button->priv->filename,
+				button->priv->size,
+				scale,
+				NULL,
+				GTK_ICON_LOOKUP_FORCE_SIZE | GTK_ICON_LOOKUP_FORCE_SVG,
+				NULL);
+
+		/*fallback to catch the case of custom icons in x11*/
+		if (!button->priv->surface && GDK_IS_X11_DISPLAY (display))
 		{
-		    button->priv->surface =
-			    panel_load_icon (button->priv->icon_theme,
+			button->priv->surface =
+				panel_load_icon (button->priv->icon_theme,
 					 button->priv->filename,
 					 button->priv->size * scale,
 					 (button->priv->orientation & PANEL_VERTICAL_MASK)   ? button->priv->size * scale : -1,
 					 (button->priv->orientation & PANEL_HORIZONTAL_MASK) ? button->priv->size * scale: -1,
 					 &error);
 		}
-		else
+		else if (!button->priv->surface)
 		{
-		    button->priv->surface =
-			    panel_load_icon (button->priv->icon_theme,
-					 button->priv->filename,
-					 button->priv->size * scale,
-					 (button->priv->orientation & PANEL_VERTICAL_MASK)   ? button->priv->size  : -1,
-					 (button->priv->orientation & PANEL_HORIZONTAL_MASK) ? button->priv->size  : -1,
-					 &error);
+			/*fallback to catch the case of custom icons not in x11*/
+			button->priv->needs_move = TRUE;
+			button->priv->surface =
+				panel_load_icon (button->priv->icon_theme,
+					button->priv->filename,
+					button->priv->size * scale,
+					(button->priv->orientation & PANEL_VERTICAL_MASK)   ? button->priv->size  : -1,
+					(button->priv->orientation & PANEL_HORIZONTAL_MASK) ? button->priv->size  : -1,
+					&error);
```

Comment:
Indents

Reply:
I spent an hour last night trying to figure out the best indent format for this. Previous code added when icons were cleared up for x11 mixed tabs and spaces. The previous code didn't provide a very good template, though I could use the same tabs and space mix it used if desired.

I can reformat all of this to spaces only in a separate PR  if needed